dependencies.yaml

```diff
@@ -373,22 +389,23 @@ dependencies:
           - matrix:
               cuda: "11.4"
             packages:
-              - cuda-version=11.4
               - cudatoolkit
               - &cudanvtx114 cuda-nvtx=11.4
               - &libcurand_dev114 libcurand-dev>=10.2.5.43,<=10.2.5.120
               - &libcurand114 libcurand>=10.2.5.43,<=10.2.5.120
           - matrix:
               cuda: "11.2"
             packages:
-              - cuda-version=11.2
               - cudatoolkit
                 # The NVIDIA channel doesn't publish pkgs older than 11.4 for
                 # these libs, so 11.2 uses 11.4 packages (the oldest
                 # available).
               - *cudanvtx114
               - *libcurand_dev114
               - *libcurand114
+          - matrix:  # Fallback for CUDA 11 or no matrix
+            packages:
+              - cudatoolkit
```

Comment:
I don't _think_ we need a fallback here. Usually the cases that need a fallback (empty matrix) are because it's needed for a `pyproject` that doesn't support matrix entries. https://github.com/rapidsai/dependency-file-generator/issues/46

Does something error out if this is removed? If not, let's cut it.

Reply:
The other reason we've had this fallback in the past is due to supporting a range of CUDA 11.* versions that had subtle differences in their CUDA packaging or what functionality they supported. That is also not so relevant now that we've largely dropped support for those versions.